src/Conversion/ONNXToKrnl/Quantization/QuantizeLinear.cpp

```diff
@@ -27,65 +27,61 @@ void emitQuantizationLinearScalarParameters(ConversionPatternRewriter &rewriter,
     Location loc, Operation *op, MemRefType inputType, MemRefType quantizedType,
     Value alloc, DimsExpr &allocDims, Value input, Value qMin, Value qMax,
     Value scale, Value zeroPoint, bool enableSIMD, bool enableParallel) {
-  MultiDialectBuilder<KrnlBuilder, MathBuilder> create(rewriter, loc);
+  MultiDialectBuilder<KrnlBuilder, MemRefBuilder, MathBuilder> create(
+      rewriter, loc);
 
   // Types
   Type quantizedElementType = quantizedType.getElementType();
   int64_t rank = inputType.getRank();
 
+  // Flatten the input data and outputs
+  DimsExpr inputDims, flatInputDims, flatAllocDims;
+  inputDims = allocDims; // Unput and output have the same shape.
+  Value flatInput =
+      create.mem.reshapeToFlatInnermost(input, inputDims, flatInputDims, rank);
```

Comment:
Just comment not for change: the flatInputDims and flatAllocDims are pure output. To me, the code will cleaner (but might be less efficient) if the flatInputDims is retrieved from the return of flatInput.

Reply:
Good point, let me look into it.

Reply:
I see your point, when the dims are not compile time, it will generate a different code. Because code is generated to compute the dimensions in reshapeFlatInnermost, and then code will be generated again to retrieve the dimensions from the new `flatValue`.  Thus more code will be generated and I will also have to update the lit tests. So unless you feel strongly about it, I would prefer not to change it.